mono/metadata/threads.c

```diff
@@ -234,154 +235,110 @@ mono_thread_get_abort_prot_block_count (MonoInternalThread *thread)
 	return (state & ABORT_PROT_BLOCK_MASK) >> ABORT_PROT_BLOCK_SHIFT;
 }
 
-static void
-verify_thread_state (gsize state)
-{
-	//can't have both INTERRUPT_REQUESTED_BIT and INTERRUPT_REQUEST_DEFERRED_BIT set at the same time
-	g_assert ((state & (INTERRUPT_REQUESTED_BIT | INTERRUPT_REQUEST_DEFERRED_BIT)) != (INTERRUPT_REQUESTED_BIT | INTERRUPT_REQUEST_DEFERRED_BIT));
-
-	//XXX This would be nice to be true, but can happen due to self-aborts (and possibly set-pending-exception)
-	//if prot_count > 0, INTERRUPT_REQUESTED_BIT must never be set
-	// int prot_count = (state & ABORT_PROT_BLOCK_MASK) >> ABORT_PROT_BLOCK_SHIFT;
-	// g_assert (!(prot_count > 0 && (state & INTERRUPT_REQUESTED_BIT)));
-}
-
 void
 mono_threads_begin_abort_protected_block (void)
 {
 	MonoInternalThread *thread = mono_thread_internal_current ();
 	gsize old_state, new_state;
+	int new_val;
 	do {
 		old_state = thread->thread_state;
-		verify_thread_state (old_state);
-
-		int new_val = ((old_state & ABORT_PROT_BLOCK_MASK) >> ABORT_PROT_BLOCK_SHIFT) + 1;
-
-		new_state = 0;
-		if (old_state & (INTERRUPT_REQUESTED_BIT | INTERRUPT_REQUEST_DEFERRED_BIT)) {
-			if (old_state & INTERRUPT_REQUESTED_BIT)
-				printf ("begin prot happy as it demoted interrupt to deferred interrupt\n");
-			new_state |= INTERRUPT_REQUEST_DEFERRED_BIT;
-		}
 
+		new_val = ((old_state & ABORT_PROT_BLOCK_MASK) >> ABORT_PROT_BLOCK_SHIFT) + 1;
```

Comment:
Well it doesn't have any particular meaning. If there is a pending sync interruption, it will be handled once we check for the interruption request flags. Although I don't think that this case is possible since after we set a sync interruption, we should be handling it pretty soon and not run more code.